ocs_ci/ocs/longevity.py

```diff
@@ -0,0 +1,704 @@
+        # Construct bulk PVC creation yaml for kube job
+        pvc_dict_list = self.construct_stage_builder_bulk_pvc_creation_yaml(
+            num_of_pvcs=num_of_pvc, pvc_size=pvc_size
+        )
+        # Construct kube job with the pvc dict list
+        pvc_job_file_list = self.construct_stage_builder_kube_job(
+            obj_dict_list=pvc_dict_list,
+            namespace=namespace,
+            kube_job_name=kube_job_name,
+        )
+        # Create stage builder for PVC kube job
+        self.create_stage_builder_kube_job(
+            kube_job_obj_list=pvc_job_file_list, namespace=namespace
+        )
+        # Validate PVCs in kube job reached BOUND state
+        pvc_bound_list = self.validate_pvc_in_kube_job_reached_bound_state(
+            kube_job_obj_list=pvc_job_file_list,
+            namespace=namespace,
+            pvc_count=num_of_pvc,
+        )
+
+        return pvc_bound_list, pvc_job_file_list
+
+    def construct_stage_builder_bulk_pod_creation_yaml(self, pvc_list, namespace):
+        """
+        This function constructs bulks pod.yamls to create bulk pods using kube_jobs.
+
+        Args:
+            pvc_list (list): List of PVCs
+            namespace (str): Namespace where the resource has to be created
+
+        Returns:
+            pods_dict_list (list): List of all Pod.yaml dict list
+
+        """
+        pods_dict_list = []
+        log.info("Constructing bulk pod creation yaml for the list of PVCs ")
+        # Get pvc objs from namespace
+        pvc_objs = get_pvc_objs(pvc_names=pvc_list, namespace=namespace)
+        for pvc_obj in pvc_objs:
+            if pvc_obj.backed_sc == constants.DEFAULT_STORAGECLASS_RBD:
+                pod_dict = templating.load_yaml(constants.CSI_RBD_POD_YAML)
+                if pvc_obj.get_pvc_vol_mode == "Block":
+                    temp_dict = [
+                        {
+                            "devicePath": constants.RAW_BLOCK_DEVICE,
+                            "name": pod_dict.get("spec")
+                            .get("containers")[0]
+                            .get("volumeMounts")[0]
+                            .get("name"),
+                        }
+                    ]
+                    del pod_dict["spec"]["containers"][0]["volumeMounts"]
+                    pod_dict["spec"]["containers"][0]["volumeDevices"] = temp_dict
+            elif pvc_obj.backed_sc == constants.DEFAULT_STORAGECLASS_CEPHFS:
+                pod_dict = templating.load_yaml(constants.CSI_CEPHFS_POD_YAML)
+            pod_name = create_unique_resource_name("test", "pod")
+            pod_dict["metadata"]["name"] = pod_name
+            pod_dict["metadata"]["namespace"] = namespace
+            pod_dict["spec"]["volumes"][0]["persistentVolumeClaim"][
+                "claimName"
+            ] = pvc_obj.name
+            pods_dict_list.append(pod_dict)
+
+        return [pods_dict_list]
```

Comment:
why again list?

Reply:
pods_dict_list contains the list of all pod yamls. If I am not returning list of list, construct_stage_builder_kube_job() is creating kubejob for each pods but I need to create a single kubejob with all the pod yamls